src/NuGet.Core/NuGet.ProjectModel/TargetFrameworkInformation.cs

```diff
@@ -143,9 +135,170 @@ public bool Equals(TargetFrameworkInformation other)
                    StringComparer.OrdinalIgnoreCase.Equals(TargetAlias, other.TargetAlias);
         }
 
-        public TargetFrameworkInformation Clone()
+        /// <summary>
+        /// Returns copy of this with specified versions defined in the Central package versions management file. 
+        /// </summary>
+        public TargetFrameworkInformation WithCentralPackageVersions(ImmutableDictionary<string, CentralPackageVersion> versions)
```

Comment:
Similar to some of the others, we probably don't need this used in more than 1 place. 
The places where we're generating the PackageSpec, such as LegacyPackageReferenceProject, VSNominationUtilities, MSBuildStaticGraphRestore and MSBuildRestoreUtility.

Reply:
Removed all With* operations and replaced with manual calls to a copy constructor